projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.spec.ts

```diff
@@ -0,0 +1,210 @@
+      expect(testHostComponent.inputValueComponent.form.value.linkValue).toEqual('Sierra');
+      expect(testHostComponent.inputValueComponent.options.length).toEqual(1);
+      expect(testHostComponent.displayInputVal.linkedResource.type).toEqual('http://0.0.0.0:3333/ontology/0001/anything/v2#Thing');
+      expect(valueInputNativeElement.readOnly).toEqual(true);
+
+    });
+    it('should make a link value editable', () => {
+
+      testHostComponent.mode = 'update';
+
+      testHostFixture.detectChanges();
+      expect(testHostComponent.inputValueComponent.mode).toEqual('update');
+
+      expect(valueInputNativeElement.readOnly).toEqual(false);
+
+      expect(testHostComponent.inputValueComponent.form.valid).toBeFalsy();
+
+      expect(testHostComponent.inputValueComponent.form.value.linkValue).toEqual('');
+      expect(testHostComponent.inputValueComponent.options.length).toEqual(1);
+
+    });
+    it('should search for resources by their label', () => {
+      const valuesSpy = TestBed.get(KnoraApiConnectionToken);
+      valuesSpy.v2.search.doSearchByLabel.and.callFake(
+        () => {
+          const res = new ReadResource();
+          res.id = 'http://rdfh.ch/0001/IwMDbs0KQsaxSRUTl2cAIQ';
+          res.label = 'hidden thing';
+          return of([res]);
+        }
+      );
+
+      testHostComponent.inputValueComponent.searchByLabel('thing');
+
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent.inputValueComponent.options.length).toEqual(1);
+
+      expect(testHostComponent.inputValueComponent.options[0].id).toEqual('http://rdfh.ch/0001/IwMDbs0KQsaxSRUTl2cAIQ');
+    });
+
+    it('should validate an existing value with an added comment', () => {
+
+      testHostComponent.mode = 'update';
+
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent.inputValueComponent.mode).toEqual('update');
+
+      expect(valueInputNativeElement.readOnly).toEqual(false);
+
+      expect(testHostComponent.inputValueComponent.form.valid).toBeFalsy();
+
+      commentInputNativeElement.value = 'this is a comment';
+
+      commentInputNativeElement.dispatchEvent(new Event('input'));
+
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent.inputValueComponent.form.valid).toBeTruthy();
+
+      const updatedValue = testHostComponent.inputValueComponent.getUpdatedValue();
+
+      expect(updatedValue instanceof UpdateLinkValue).toBeTruthy();
+
+      expect((updatedValue as UpdateLinkValue).valueHasComment).toEqual('this is a comment');
+
+    });
+    it('should return a selected resource', () => {
+
+      const res = new ReadResource();
+      res.id = 'http://rdfh.ch/0001/a-blue-thing';
+      testHostComponent.inputValueComponent.setLinkValue(res);
+
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent.inputValueComponent.linkedResourceIRI).toEqual('http://rdfh.ch/0001/a-blue-thing');
+    });
```

Comment:
Could you please add tests for the value creation case?

Reply:
yes, just doing that.